src/azure-cli-core/azure/cli/core/commands/validators.py

```diff
@@ -71,13 +71,13 @@ def generate_deployment_name(namespace):
 def get_default_location_from_resource_group(cmd, namespace):
     if not namespace.location:
         from azure.cli.core.commands.client_factory import get_mgmt_service_client
-        from msrestazure.azure_exceptions import CloudError
+        from azure.core.exceptions import HttpResponseError
         from knack.util import CLIError
 
         resource_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
         try:
             rg = resource_client.resource_groups.get(namespace.resource_group_name)
-        except CloudError as ex:
+        except HttpResponseError as ex:
```

Comment:
Perhaps we don't use try catch here to let `azure.cli.core.parser.AzCliCommandParser.validation_error` handle exceptions, such as `azure.core.exceptions.ResourceNotFoundError`?

Reply:
OK, makes sense~
Here raise the original specific error seems to be a better choice．

Reply:
As we discussed offline, we first maintain the latest exit code and error throwing ways (because it seems to work well at present), so we remove the `try-catch` logic first.

If additional information is needed to help locate the problem in the future, we can consider recording the information in the form of error log